### ISSUE TITLE
Feature proposal: Mermaid diagrams

### ISSUE THREAD

**Issue by marcosh** [haskell/haddock#1581]
When generating `html` documentation, it would be nice to automatically render Mermaid diagrams.

For example, if some Haskell code contained a comment like

```haskell
-- | Some docs with a diagram
-- ```mermaid
-- flowchart LR
--    Start --> Stop
-- ```
```

should render as

> Some docs with a diagram
```mermaid
flowchart LR
   Start --> Stop
```

**Comment by Kleidukos** [comment]
Yes, I had the same idea for a Haddock plugin. Many new things will unlock when we can rework the architecture of Haddock in the near future, so thank you for opening this ticket. :)

**Comment by Kleidukos** [comment]
I'm closing this for now. Haddock has migrated to [Gitlab](https://gitlab.haskell.org/ghc/ghc/-/tree/master/utils/haddock?ref_type=heads).
